scale/product/views.py

```diff
@@ -38,13 +38,24 @@ def list(self, request):
         is_operational = rest_util.parse_bool(request, 'is_operational', required=False)
         is_published = rest_util.parse_bool(request, 'is_published', default_value=True)
         file_name = rest_util.parse_string(request, 'file_name', required=False)
+        job_output = rest_util.parse_string(request, 'job_output', required=False)
+        recipe_ids = rest_util.parse_int_list(request, 'recipe_id', required=False)
+        recipe_type = rest_util.parse_string(request, 'recipe_type', required=False)
+        recipe_job = rest_util.parse_string(request, 'recipe_job', required=False)
+        batch_ids = rest_util.parse_int_list(request, 'batch_id', required=False)
+
+        source_started = rest_util.parse_timestamp(request, 'source_started', required=False)
+        source_ended = rest_util.parse_timestamp(request, 'source_ended', required=False)
+        rest_util.check_time_range(source_started, source_ended)
```

Comment:
Remove source_started and source_ended and instead use a time_field parameter for indicating which time field to apply started and ended to.